BEST-PRACTICES.md

```diff
@@ -0,0 +1,49 @@
+# Best Practices
+
+We're glad you're considering contributing to SUSHI! Below are a few best practices that we recommend for all contributions.
+
+## Before contributing
+
+Before contributing a feature or a bugfix, we recommend creating a GitHub issue if one does not exist. This allows the community to provide feedback on why an issue may be occurring or provide additional insight into a suggested feature.
+
+If a GitHub issue already exists for what you are planning to contribute, we recommend commenting on the issue to indicate that you are working on an implementation to avoid duplication of work.
+
+## Coding practices
+
+We recommend the following coding practices for high quality contributions:
+
+- Make all changes in a personal [fork](https://help.github.com/articles/fork-a-repo/) of this repository.
+- Use descriptive commit messages.
+- Prefer self-explanatory code as much as possible, but provide helpful comments for complex expressions and code blocks.
+- Add unit tests for any new or changed functionality, and update any existing tests that are impacted by your changes.
+  - SUSHI uses [Jest](https://jestjs.io/) as a testing framework.
+  - To run the full test suite, run `npm test`. Ensure all tests are passing.
+- Follow the code style and conventions as enforced by the lint configuration and as evidenced by the existing code.
+  - SUSHI uses [ESLint](https://eslint.org/) for code linting.
+  - To run the linter on all code, run `npm run lint`. Ensure there are no issues reported.
+- Follow the code formatting as enforced by the formatter configuration.
+  - SUSHI uses [Prettier](https://prettier.io/) for code formating.
+  - To run the prettier on all code, run `npm run prettier`. Ensure there are no issues reported.
+- Ensure any new dependencies do not contain any known security vulnerabilities
+  - To check for known security vulnerabilities, we recommend using [npm-audit](https://fshschool.org/). Run `npm audit` and ensure there are no new issues on your branch.
```

Comment:
Can we also indicate that some audit issues can be fixed using `npm audit fix`? And if a vulnerability does not have an existing patch, or is not feasible to address for some other reason, to document it in DEPENDENCY-NOTES.md?

Reply:
Hm, I'm not sure if I agree with this one. I think it has been pretty rare that we haven't been able to address issues reported in `npm audit`, so I'm hesitant to document in a best practices document a way to avoid addressing them.

I do think that including the `DEPENDENCY-NOTES.md` file is a good idea. I think I lean toward including another bullet point to say "Ensure any new dependencies use the latest published version". Under that bullet point, I think it makes sense to include a note that if this isn't possible, document it in the `DEPENDENCY-NOTES.md` file. What do you think?

Reply:
OK. I like your suggestion.

Reply:
Done in [db172dd](https://github.com/FHIR/sushi/pull/1474/commits/db172dd4017b281678ddefe358e874e493770c89).